### DISCUSSION TITLE
Multilangual documents

### DISCUSSION THREAD

**Discussion by ChinaeduO** [outline/outline#6126] [Ideas]
In would be very nice to have the ability to translate documents. So that one document can be edited in multiple languages. In addition, a DeepL integration would cherry top it.

**Comment by tommoor** [comment]
I would like to do something here, especially given how good the translation support is for the Product UI.

Do you have thoughts on how this would work?

**Reply by ChinaeduO** [reply]
Like technically or visually ?

**Reply by tommoor** [reply]
In terms of user experience

**Reply by microbearlogist** [reply]
This would certainly be a key function for the already amazing Outline wiki!


Personally, I'd see it like this:


> **1. For Editor**

Add the possibility of creating a version of a document in another language

Meaning two additional entries in the document's main menu:
Ideally in a separate category between the "Edit/Pin" block and the "Comment/Print" block.

- Add another language
- Define the language of this version

"Add another language" asks you to define the language of this new version from a dropdown menu (showing language supported by  Outline), then, by default, the initial document is duplicated to facilitate faithful translation.

When the document exists in several languages, a "language" menu icon appears next to the "Hide content" and "Star content" menus. When clicked, it displays a drop-down menu for selecting the document language. Ideally, this language icon should display the initials of the chosen language next to it (like "EN", "DE", "FR"...) or a flag (see which is more legible). 
This menu is used to navigate between language versions.

Alternatively (But that could make things quickly unreadable with a lot of doc), multilingual versions of a document could be displayed in the same way as "Nested documents" in the left navigation menu. But grayed out and with the language specified in (parenthesis). And only when edition mode is activated.



> **2. For Readers:**

- The language navigation menu (same as for the editor).
- Optional function (can be enabled/disabled in preferences): If the document exists in the language of the user interface, then it's the one displayed by default (unless the admin decides to force a default language).


> **For Admin :**

- The ability to force a default document language on the entire wiki.
(This is to force a reference version, in case the translations are not totally reliable).


I can make a small series of images to illustrate this if that helps @tommoor

**Reply by paulux84** [reply]
We are managing multiple languages using different collections, but this type of management is really difficult when links to other doc element are added inside the main language because all links must be substituted with links of translated language version.

**Reply by tommoor** [reply]
Do you actually _want_ folks to be translating the same document into multiple languages? Given how good LLM's have gotten at this task it seems unnecessary and a massive amount of maintenance to rely on humans to keep multiple versions of a document uptodate.

It really feels like this feature would be better implemented through real-time AI translation – define a "primary" language and secondary language and have the secondary ones kept in-sync automatically..

**Reply by miikka-virtanen** [reply]
Yes, I'd rather have people translate the documents. They can still use AI as a tool for this if they'd like, but implementing translations _only_ through real time AI translations seems the wrong way to go about this. AI translations are not perfect, especially not with less common languages (like Finnish) or documents with highly specific business terms (which do not have obvious translations), and you'd have no control over the results.

I'd rather see this implemented as a generic feature that is agnostic to where the translations come from, and then further AI integrations for those that want them. To help keep language versions up to date, Outline could flag translated versions as "out of date" when the primary language document is updated.

**Reply by thumDer** [reply]
Currently we have both official languages in the same doc, one below the other, but this meens duplicate embedded images, and usually the second language is not updated. We would also really like the feature of built in support for multiple languages, just like it was described above. Regarding the real-time AI I think a combination of default AI translations with the possibility to replace them with a manual one, where the UI shows wether it is an AI translation or a manual one could work.

**Reply by labor4** [reply]
we're doing very technical pages, and I wouldn't even try to aim AI at it.
too specific.

**Comment by VeniceNerd** [comment]
Has there been made any update on this? I would love to be able to have different language versions of each page.

**Comment by simonv3** [comment]
Feels like as for implementation, gitbook's content variants might be something to look at? https://gitbook.com/docs/publishing-documentation/site-structure/variants. 

I'm new to the codebase but looking at your models, it feels like you could introduce a DocumentVariant model that's a one to many from Document, which in this instance would be the main language. Looking at gitbook, this might be cool to document versions too (but not our use case). 

We're looking at outline and it seems to be much easier to grok and teach than gitbook, but we have a pretty multi-lingual community so this would be a big boon for us, and I think we can make do with collections for now, but it would be great to have this be a first level experience for each doc and I'm hoping it's not a thing of "well we'd have to entirely re-architect the app to make this happen".

**Comment by FreeSynergy** [comment]
I would like the multilanguage too. This is necessary for "Wikis" in decentral Networks. I want to change from "wiki.js" to outline, because there are some issues (eg. links don't get updated, when you move one Document to another place. Or there is no standard to embed YouTube movies and so on). Or easier: It's outdated and there is no continue anymore - just bugfixes …

The features of Outline, with multi-language support and some other small enhancements would bring this software to the top.
It's fast, modern and very good …

Please, put in this feature …

**Comment by arslc** [comment]
Yes, I agree! This feature is really missing. I hope someone can implement it soon.

@tommoor what do you think about this?

**Comment by MystikReasons** [comment]
+1 for me

**Comment by fabianfreund** [comment]
+1 for me too

**Comment by ValleBL** [comment]
+1 for more language improvements like auto translate based on your profile language or even a DeepL integration

**Comment by gmglbn-0** [comment]
+1 for me

**Comment by Nachichuri** [comment]
+1 too

**Comment by clarity-marius** [comment]
same here

**Comment by PrzemyslawSochon** [comment]
+1 folks
